homeassistant/components/backup/agent.py

```diff
@@ -31,27 +37,25 @@ class BackupAgent(abc.ABC):
     async def async_download_backup(
         self,
         backup_id: str,
-        *,
-        path: Path,
         **kwargs: Any,
-    ) -> None:
+    ) -> BackupAgentStream:
         """Download a backup file.
 
         :param backup_id: The ID of the backup that was returned in async_list_backups.
-        :param path: The full file path to download the backup to.
+        :return: Either an aiohttp or asyncio StreamReader.
         """
 
     @abc.abstractmethod
     async def async_upload_backup(
         self,
         *,
-        path: Path,
+        open_stream: Callable[[], Coroutine[Any, Any, AsyncGenerator[bytes]]],
```

Comment:
Maybe `open_stream` is not a good name for this parameter

Reply:
I think it's ok.